source/simulator/solver.cc

```diff
@@ -853,6 +863,12 @@ namespace aspect
                                    solver_control_cheap,
                                    solver_control_expensive);
 
+        // We do not need the linearized_stokes_initial_guess vectory anymore, so
+        // (ab)use it for computing the initial residual when the Newton solver is used
```

Comment:
Can you extend this comment a bit. Specifically, I do not quite understand why we compute the initial residual before the solution for the fixed point iterations, but for the Newton iterations we compute it after the solution.

Reply:
I put more comments for both the newton and non-Newton residual computation. I hope this helps a bit.